flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -242,6 +248,86 @@ public void start() throws Exception {
 	}
 
 	@Override
+	public CompletableFuture<JobStatusesWithIdsOverview> requestJobIdsOverview(@RpcTimeout Time timeout) {
```

Comment:
I think it would be better to define `requestJobsOverview` which simply returns a `Collection<JobOverview>` with `JobOverview` containing the `JobID` and the `JobStatus`, for example. All other data processing and data restructuring like grouping the individual jobs to make it easier for the Web UI to display the content, should then happen on the handler and not in the `Dispatcher`. My gut feeling is that this is unnecessary coupling of unrelated components.

Reply:
I think that makes a lot of sense. Will address this.